dask/order.py

```diff
@@ -133,6 +134,14 @@ def order(dsk, dependencies=None):
     # tree, we skip processing it normally.
     # See https://github.com/dask/dask/issues/6745
     root_nodes = {k for k, v in dependents.items() if not v}
+    if len(root_nodes) > 1:
+        root = "root-node"
+        dsk[root] = (object(), *root_nodes)
+        dependencies[root] = root_nodes
+        o = order(dsk, dependencies)
+        del o[root]
+        return o
```

Comment:
This is by far the most interesting change in this PR since it is effectively disabling a lot of logic. For one, this is disabling the entire `skip_root_node` and also specialized `finish_now` things.

What's even more critical is that this effectively disables many root-node dependent graph metrics (stuff like max_dependencies, min_dependencies) since all nodes now share the same goal

Reply:
There is an interesting effect, though. The partition_keys function is depending on `min_dependencies`, or rather the difference between `min_dependencies` and `total_dependencies`. If min_dependencies was removed entirely, this function would become negative, reverting every decision.